### PR TITLE
Introduce ApprovalTests.Net

### LINKED ISSUE #164: Introduce gold master testing
For now, we have testing code like that:
```fsharp
let expected = (create-expected-code)
let actual = parse "some formula"
assertDeepEqual expected actual
```

While that's okay, `assertDeepEqual` often produces unreadable output. E.g. now I have this:

```
DeepEqual.Syntax.DeepEqualException
Comparison Failed: The following 10 differences were found.
	Actual.RootAtom.RowAtom != Expected.RootAtom.RowAtom (WpfMath.Atoms.RowAtom != "(missing)")
	Actual.RootAtom.Background != Expected.RootAtom.Background ((null) != "(missing)")
	Actual.RootAtom.Foreground != Expected.RootAtom.Foreground (#FFED1B23 != "(missing)")
	Actual.RootAtom.PreviousAtom != Expected.RootAtom.PreviousAtom ("(missing)" != (null))
	Actual.RootAtom.Elements != Expected.RootAtom.Elements ("(missing)" != System.Collections.ObjectModel.ReadOnlyCollection`1[WpfMath.Atoms.Atom])
	Actual.<RootAtom>k__BackingField.RowAtom != Expected.<RootAtom>k__BackingField.RowAtom (WpfMath.Atoms.RowAtom != "(missing)")
	Actual.<RootAtom>k__BackingField.Background != Expected.<RootAtom>k__BackingField.Background ((null) != "(missing)")
	Actual.<RootAtom>k__BackingField.Foreground != Expected.<RootAtom>k__BackingField.Foreground (#FFED1B23 != "(missing)")
	Actual.<RootAtom>k__BackingField.PreviousAtom != Expected.<RootAtom>k__BackingField.PreviousAtom ("(missing)" != (null))
	Actual.<RootAtom>k__BackingField.Elements != Expected.<RootAtom>k__BackingField.Elements ("(missing)" != System.Collections.ObjectModel.ReadOnlyCollection`1[WpfMath.Atoms.Atom])
   at DeepEqual.Syntax.ObjectExtensions.ShouldDeepEqual(Object actual, Object expected, IComparison comparison)
   at WpfMath.Tests.Utils.assertParseResult(String formula, TexFormula expected) in X:\MyDir\Projects\wpf-math\src\WpfMath.Tests\Utils.fs:line 41
   at WpfMath.Tests.ParserTests.\color should parse arguments properly(String text) in X:\MyDir\Projects\wpf-math\src\WpfMath.Tests\ParserTests.fs:line 188
```

Not very readable!

I suggest we use the "[Gold Master Testing](https://en.wikipedia.org/wiki/Characterization_test)" technique instead (for the parser tests at least). It means something like that:

```fsharp
let actual = parse "some formula"
serialize actual "actual.json"
compare "expected.json" "actual.json"
```

Where `serialize` should save the parse tree e.g. as JSON (or any other text, actually), and `compare` runs file diff and shows the readable diff result. Some IDEs offer additional support for gold master testing (e.g. they will show a file diff in a proper IDE window with highlighting and stuff, very conveniently): I know for fact that Rider supports that, because this is used for Rider tests internally.

In case of a first time run (or in case of changes in the test output), gold master tests usually include a special mode to create initial `expected.json` file.

**Pros**:

1. Simpler to prepare new tests.
2. Simpler to support tests in case of changes.
3. Simpler to look for subtle changes such as `SourceSpan` coordinates: currently we've disabled the `SourceSpan` comparison for many tests because it's too hard to support that.

**Cons**:

1. It becomes simple to change a lot of tests, so it will be easier to introduce small errors; we'll need to review test data as well as test code.
2. Need to migrate to this new test mode.

### PR DESCRIPTION
**Pull request by ForNeVeR**
## Description
I've introduced the new tests in the following main steps:

1. Verify all the data in the existing tests twice (verify the parsing result, then verify the programmatically constructed expected value, and also check that they are the same): commit b39b484.
2. Remove the code generating the expected results (don't more needed): commit 080ef7a.
3. Refactor what's left.

## TODO
- [x] Sort out `SourceDetectionTests.fs`
- [x] Move `src` function
- [x] Rename the test cases so the corresponding file names don't look so scary

## Impact
Closes #164.

### CODE REVIEW

File: src/WpfMath.Tests/ApprovalTestUtils.fs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+[<assembly: UseReporter(typeof<DiffReporter>)>]
+[<assembly: UseApprovalSubdirectory("TestResults")>]
+do ()
+
+type private InnerPropertyContractResolver() =
+    inherit DefaultContractResolver()
+    member private __.DoCreateProperty(p, ms) =
+        let prop = base.CreateProperty(p, ms)
+        prop.Readable <- true
```
Comment (gsomix) [inline src/WpfMath.Tests/ApprovalTestUtils.fs]: IIRC you can init properties in ctor.

```fsharp
let prop = base.CreateProperty(p, ms, Readable = true)
```

Comment (ForNeVeR) [reply src/WpfMath.Tests/ApprovalTestUtils.fs]: Wait, really? It works with functions and not only with real "constructors"? Wow.

Comment (gsomix) [reply src/WpfMath.Tests/ApprovalTestUtils.fs]: @ForNeVeR Oh, not works. :( My bad, need to sleep more.

Comment (ForNeVeR) [reply src/WpfMath.Tests/ApprovalTestUtils.fs]: Wait, it works actually

**Comment by ForNeVeR** [comment]
Thanks for review.

**Pull request merged**